tests/test_scm.py

```diff
@@ -24,7 +54,130 @@ def test_commit(self):
         G = Git(self._root_dir)
         G.add(["foo"])
         G.commit("add")
-        self.assertTrue("foo" in self.git.git.ls_files())
+        self.assertIn("foo", self.git.git.ls_files())
+
+    def test_open(self):
+        G = Git(self._root_dir)
+
+        from dvc.utils.compat import open
+
+        # XXX: this is wrong way to deal with encodings, but it
+        # looks working for both py2/py3 in DVC case
+        with open(self.FOO, "w", encoding="utf-8") as f:
+            f.write("проверка\n")
+
+        G.add([self.FOO, self.DATA_DIR])
+        G.commit("add")
+        self.assertEqual(G.open(self.FOO).read(), "проверка\n")
+        with self.assertRaises(IOError):
+            G.open("not-existing-file")
+        with self.assertRaises(IOError):
+            G.open("not-existing-file", branch="master")
+        with self.assertRaises(IOError):
+            G.open(self.DATA_DIR, branch="master")
+
+    def test_exists(self):
+        G = Git(self._root_dir)
+        self.assertTrue(G.exists(self.FOO))
+        self.assertFalse(G.exists(self.FOO, branch="master"))
+        self.assertFalse(G.exists("non-existing-file"))
+        self.assertFalse(G.exists("non-existing-file", branch="master"))
+        G.add(["foo"])
+        G.commit("add")
+        self.assertTrue(G.exists(self.FOO, branch="master"))
+
+    def test_isdir(self):
+        G = Git(self._root_dir)
+        self.assertTrue(G.isdir(self.DATA_DIR))
+        self.assertFalse(G.isdir(self.DATA_DIR, branch="master"))
+        self.assertFalse(G.isdir(self.FOO))
+        G.add([self.FOO, self.DATA_DIR])
+        G.commit("add")
+        self.assertTrue(G.isdir(self.DATA_DIR, branch="master"))
+        self.assertFalse(G.isdir(self.FOO, branch="master"))
+
+
+class AssertWalkEqualMixin(object):
+    def assertWalkEqual(self, actual, expected):
```

Comment:
Is there need for duplication of this method (first occurence is in TestDir)?
I think we could:
- make this method static in  TestDir
or
- move it to `tests/utils`

Reply:
Good catch, thanks. As of my sense it is better to keep it in the Mixin, because `assertWalkEqual` is needed only in the  tests of `walk()` method implementations.

Moving to tests/utils... imho when the code is needed in a single place, then it is better to keep this code close to the place where it is needed. Maybe we should move `Test*Walk` cases into the separate file together with this mixin?

And what's the point of making it static?

Reply:
I agree with keeping in one place, now its my bad that I assumed both are used. 

There is no point in making this static, my bad.